### ISSUE TITLE
the way we're doing strings isn't very DRY

### ISSUE THREAD

**Issue by johngruen** [mozilla/testpilot#699]
Do we need strings in the templates or just in the l20n file? It's a pain to track down strings in content and then duplicate the the l20n string.

**Comment by lmorchard** [comment]
They have to appear in the l20n file(s) - otherwise you just get the `data-l10n-id` value. It doesn't hurt anything to leave them in the templates, though. We could delete them as we go, but they also function a bit as comments in the templates so we know what various things are, rather than just empty elements.

**Comment by johngruen** [comment]
@pdehaan seems like fxa does something clever here. Maybe extracting strings from the templates to create the English l10n file. Could we do something like that?

**Comment by lmorchard** [comment]
I think FxA uses gettext [for localization](https://github.com/mozilla/fxa-content-server-l10n), which has tools for string extraction. L20N doesn't have such tools yet, as far as I could see. It's a manual process

**Comment by pdehaan** [comment]
I asked in #l20n on IRC and the response was basically just don't put the strings in the content:

> 9:45 AM <pdehaan> is there an l20n CLI tool for extracting strings from HTML templates and generating a .l20n file automatically?  We seem to have a bunch of string duplication between our HTML templates and our l20n files.
> 9:47 AM <pdehaan> Basically trying to help solve https://github.com/mozilla/testpilot/issues/699
> 9:48 AM <Pike> in l20n, just don't put the strings into the template
> 9:49 AM <Pike> string extractions like to generate problems with duplicate strings or IDs

**Comment by lmorchard** [comment]
Yeah, so we can create templates that never have strings in the first place (i.e. just empty elements), but that seems troublesome for creating them and seeing what goes in them before pulling strings out into L20N. We could also go back and delete all text strings from templates after putting them into L20n, but that also kind of seems like busy work.

TL;DR: The DRY way here for L20n is to build templates that never contain text, just empty elements with  `data-l10n-id`s. Then, write the text only in the locale files.

**Comment by ghost** [comment]
Wontfix: sub-optimal.
